projectq/backends/_awsbraket/_awsbraket.py

```diff
@@ -0,0 +1,481 @@
+            compatibility but in AWSBraket a previously executed circuit
+            will store the results in the S3 bucket and it can be retreived
+            at any time after.
+            It should require no circuit execution at the time of retrieving
+            the results and probabilities.
+            In order to obtain the probabilities of a previous job
+            you have to get the TaskArn and remember the qubits and ordering
+            used in the original job.
+
+        Args:
+            qureg (list<Qubit>): Quantum register determining the order of the
+                qubits.
+
+        Returns:
+            probability_dict (dict): Dictionary mapping n-bit strings to
+                probabilities.
+
+        Raises:
+            RuntimeError: If no data is available (i.e., if the circuit has
+                not been executed). Or if a qubit was supplied which was not
+                present in the circuit (might have gotten optimized away).
+        """
+        if len(self._probabilities) == 0:
+            raise RuntimeError("Please, run the circuit first!")
+
+        probability_dict = dict()
+        for state in self._probabilities:
+            mapped_state = ['0'] * len(qureg)
+            for i in range(len(qureg)):
+                mapped_state[i] = state[self._logical_to_physical(qureg[i].id)]
+            probability = self._probabilities[state]
+            mapped_state = "".join(mapped_state)
+            if mapped_state not in probability_dict:
+                probability_dict[mapped_state] = probability
+            else:
+                probability_dict[mapped_state] += probability
+        return probability_dict
+
+    def _run(self):
+        """
+        Run the circuit.
+
+        Send the circuit via the AWS Boto3 SDK. Use the provided Access Key and
+        Secret key or ask for them if not provided
+        """
+        # finally: measurements (no intermediate measurements are allowed)
+        # No explicit measurements have to be writen down. To run the circuit
+        # make implicit measurement of all the qubits
+
+        # In Braket the results for the jobs are stored in S3.
+        # You can recover the results from previous jobs using the
+        # TaskArn (self._retrieve_execution).
+        # Keept the code for retreive here to maintain the same strcuture
+        try:
+            if self._retrieve_execution is not None:
+                res = retrieve(credentials=self._credentials,
+                               taskArn=self._retrieve_execution,
+                               num_retries=self._num_retries,
+                               interval=self._interval,
+                               verbose=self._verbose)
+            else:
+                # Return if no operations are added.
+                if self._circuit == "":
+                    return
+
+                n_qubit = max(self._allocated_qubits) + 1
```

Comment:
Since self._allocated_qubits set is constructed just addign to it the qubit id when the Allocate command is used, I changed max by len:

`n_qubit = len(self._allocated_qubits)`

I took this from the AQT and IBM backends, therefore IMHO it would be required to review the corresponding lines. If you think so I could do it after this PR